packages/babel-helper-environment-visitor/src/index.ts

```diff
@@ -1,38 +1,52 @@
-import type { NodePath } from "@babel/traverse";
-import { VISITOR_KEYS, staticBlock } from "@babel/types";
+import type { NodePath, Visitor } from "@babel/traverse";
 import type * as t from "@babel/types";
 
 // TODO (Babel 8): Don't export this function.
 export function skipAllButComputedKey(
   path: NodePath<t.Method | t.ClassProperty>,
 ) {
-  // If the path isn't computed, just skip everything.
-  if (!path.node.computed) {
-    path.skip();
-    return;
+  path.skip();
+  if (path.node.computed) {
+    // requeue the computed key
+    path.context.maybeQueue(path.get("key"));
   }
+}
 
-  // So it's got a computed key. Make sure to skip every other key the
-  // traversal would visit.
-  const keys = VISITOR_KEYS[path.type];
-  for (const key of keys) {
-    if (key !== "key") path.skipKey(key);
+function skipAndrequeueComputedKeysAndDecorators(
```

Comment:
```suggestion
function skipAndRequeueComputedKeysAndDecorators(
```

(or maybe something with less words like `skipInternalContext` or `skipContents` works too).

Reply:
I think `skipInternalContext` and `skipContents` are a bit vague since they are not formally defined in the spec. Maybe  `skipAndRequeueKeyAndDecorator`? I also thought about `skipParamAndBodyAndInitializer` but it seems no better than `skipAndRequeueKeyAndDecorator`.